articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -0,0 +1,235 @@
+---
+layout: default
+title: ROS QoS - Deadline, Liveliness, and Lifespan
+permalink: articles/qos_deadline_liveliness_lifespan.html
+abstract:
+  This article makes the case for adding Deadline, Liveliness, and Lifespan QoS settings to ROS. It outlines the requirements and explores the ways it could be integrated with the existing code base. 
+author: '[Nick Burek](https://github.com/nburek)'
+published: true
+categories: Middleware
+date: February 13th 2019
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+Glossary:
+
+- DDS - Data Distribution Service
+- RTPS - Real-Time Publish Subscribe
+- QoS - Quality of Service
+- Service Client - Also referred to as just Client, refers to an application that connects to a ROS Service to send requests and receive responses.
+- Service Server - Also referred to as just Server, refers to the application that is running a ROS Service that receives requests and sends responses.
+
+## Existing ROS QoS Settings
+
+While DDS provides many settings to enable fine-grained control over the Quality of Service (QoS) for entities, ROS only provides native support for a handful of them.
+ROS users are able to specify the History, Depth, Reliability, and Durability via a QoS configuration struct when they create Publishers, Subscribers, etc.
+
+This leaves a lot of QoS settings that can only be set if DDS vendor can load additional default settings via a configuration file.
```

Comment:
With QoS setting we are very DDS specific. What is it expected from the non-DDS rmw implementations. That they replicate all QoS settings or that they do not implement them?

Reply:
This is something I'd like to hear about from [people who have implemented alternative `rmw`s](https://discourse.ros.org/t/non-dds-based-rmw-implementation/5890).

Reply:
@dejanpan What we had discussed was to define these QoS settings in terms of ROS system behavior without the assumption of a DDS based implementation of rmw. The expectation in these cases is a non DDS based implementation of rmw would have to implement them to support them, but that we would also develop a mechanism for them to inform the application in cases where they request a policy that is not supported by the rmw implementation. So your rmw implementation doesn't have to support every QoS setting and it would be up to the application to decide what to do if the rmw implementation doesn't support a desired policy.